### ISSUE TITLE
Feature request: bundle plpythonu 

### ISSUE THREAD

**Issue by davidfilat** [PostgresApp/PostgresApp#497]
Hi,
I love the app! It's the best tool yet for a lightweight postgres server on my local computer!
But I would really love to have plpython included in the app, or at least some instructions on how can it be added. Do you think it would be possible to bundle it in the app?
Thank you!

**Comment by jakob** [comment]
It's included. You can enable the extension using:
```sql
create extension plpythonu;
```

Then you can test if it works like this:
```sql
do language plpythonu $$
	plpy.notice("It works!")
$$;
```

It's Python 2 only, unfortunately.

**Comment by davidfilat** [comment]
> It's included. You can enable the extension using:
> 
> ```sql
> create extension plpythonu;
> ```
> 
> Then you can test if it works like this:
> 
> ```sql
> do language plpythonu $$
> 	plpy.notice("It works!")
> $$;
> ```
> 
> It's Python 2 only, unfortunately.

Cool! Do you plan on adding python3 support?

**Comment by jakob** [comment]
Not at the moment. macOS only comes with Python 2, so I think we would need to include Python 3 with Postgres.app, which I'm not sure how to do.

**Comment by alitrack** [comment]
@davidfilat 
please have a look
[How to enable plpython3u extension on Postgre.app](http://blog.rubypdf.com/2019/06/29/how-to-enable-plpython3u-extension-on-postgre-app/)

**Comment by tonytheodore** [comment]
Catalina has Python 3.7.3. However, from what I've read, the version of python at runtime should be the same as the build time version or strange things can happen. I already have four `python3` binaries (apple, homebrew, macports, and anaconda) and now am ready to start using 3.8 which came out a few weeks ago.

As much as I also want this feature, I don't see how it can be done "safely" without also bundling python. Then you have the problem of which python to start with and how does one install python libraries? You'd almost end up duplicating a lot of python packaging efforts and it would be a major distraction.

I think python users probably need to build/obtain their own extensions. The [method above](http://blog.rubypdf.com/2019/06/29/how-to-enable-plpython3u-extension-on-postgre-app/) is a great starting point.

**Comment by jakob** [comment]
@tonytheodore thanks for your comment. I did not know Apple shipped python3 with Catalia. It would be really nice if we could ship plpython3u with Postgres.app.

However, Apple did mention that scripting languages shipped with the OS are deprecated and might not be included anymore at all with macOS 10.16. At the same time they seem to recommend switching to python3, so I'm not sure what this means for Postgres.app in the future.

I really don't want to start bundling scripting languages with Postgres.app (we do include plv8, and that's already a mess...).

I wonder if it's possible to build the pl/python extension in a way so it can interface with different versions of python, or maybe we should just support one distribution (eg. require that you install the packages from Python.org if you want pl/python3 support?)

**Comment by davidfilat** [comment]
> @tonytheodore thanks for your comment. I did not know Apple shipped python3 with Catalia. It would be really nice if we could ship plpython3u with Postgres.app.
> 
> However, Apple did mention that scripting languages shipped with the OS are deprecated and might not be included anymore at all with macOS 10.16. At the same time they seem to recommend switching to python3, so I'm not sure what this means for Postgres.app in the future.
> 
> I really don't want to start bundling scripting languages with Postgres.app (we do include plv8, and that's already a mess...).
> 
> I wonder if it's possible to build the pl/python extension in a way so it can interface with different versions of python, or maybe we should just support one distribution (eg. require that you install the packages from Python.org if you want pl/python3 support?)
> 
> 

@jakob I really like the idea and it would be awesome if you could integrate the plpython3u extension with the user default python distribution (maybe limiting it to the homebrew or anaconda distribution will make the task easier)
